lib/util.js

```diff
@@ -186,19 +195,30 @@ function inspect(obj, opts) {
     exports._extend(ctx, opts);
   }
   // set default options
-  if (ctx.showHidden === undefined) ctx.showHidden = false;
-  if (ctx.depth === undefined) ctx.depth = 2;
-  if (ctx.colors === undefined) ctx.colors = false;
-  if (ctx.customInspect === undefined) ctx.customInspect = true;
-  if (ctx.showProxy === undefined) ctx.showProxy = false;
+  Object.keys(inspectOptions).forEach(function(option) {
```

Comment:
Looking at the code, `ctx` contains the user options. Wouldn't `Object.assign(ctx, inspectOptions)` overwrite the user options with the default options?